.github/actions/detect-workflow/action.yml

```diff
@@ -0,0 +1,31 @@
+name: 'Detect Reusable Workflow Repository and Ref'
+description: 'Get the current reusable workflow repository name and ref. Requires "id-token: write" permission'
+outputs:
+  repository:
+    description: The current workflow repository, format org/repository
+    value: ${{ steps.detect.outputs.repository }}
+  ref:
+    description: The current workflow reference
+    value: ${{ steps.detect.outputs.ref }}
+runs:
+  using: 'composite'
+  steps:
+    - run: | 
+        set -euo pipefail
+        
+        status_code=$(curl -sS -H "Authorization: Bearer $ACTIONS_ID_TOKEN_REQUEST_TOKEN" "$ACTIONS_ID_TOKEN_REQUEST_URL&audience=slsa-framework/slsa-github-generator-go/builder" -o jwt.json -w '%{http_code}')
+        if [[ $status_code -ge 400 ]]; then
```

Comment:
let's check `$status_code <200 || $status_code >= 300`, since all 2xx should be acceptable.
(the 400 is my fault, I copied this code without checking what it was doing)

Reply:
updated this with Ian's go code, which has this check